core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -849,4 +850,28 @@ private String toStringSlow(@Nullable Channel ch, @Nullable RequestLogAccess par
             return buf.toString();
         }
     }
+
+    @Override
+    public CompletableFuture<Void> initiateConnectionShutdown() {
+        final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+
+        setAdditionalRequestHeader(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        log().whenRequestComplete().whenComplete((ignore, ex) -> {
+            if (ex != null) {
+                completableFuture.completeExceptionally(ex);
+                return;
+            }
+            final Channel ch = channel();
+            assert ch != null;
```

Comment:
I wonder channel is always not null in this case.
If it's nullable, I will add below code instead of this assert.
```
if (ch == null) {
    completableFuture.complete(null);
    return;
}
```

Reply:
Channel can still be null when a client failed to send a request even before acquiring a channel from the pool, so yeah, please add it back.